api/ttn/lorawan/v3/notification_service.proto

```diff
@@ -40,7 +40,7 @@ message Notification {
   EntityIdentifiers entity_ids = 3;
 
   // The type of this notification.
-  string notification_type = 4;
+  NotificationType notification_type = 4;
```

Comment:
This is a breaking API change as it changes the wire type of this field. I think the other field should be deprecated and a new field should be introduced.

Is this part of any external API currently?

Reply:
Regarding this, the API change would be that the accepted list of values went from lowercase to uppercase since there is not prefix. Is it worth it looking into updating https://github.com/TheThingsIndustries/protoc-gen-go-json to have an option which makes the value always be upper case?

The idea being that certain API endpoints would be case insensitive, and therefore this change wouldn't break the API.

Reply:
currently this field is part of the UpdateUser endpoint and CreateUser, which are used by frontend and, I assume, in the CLI by users. So, if I understand this correctly, it would be better to keep the old field as deprecated and add a new one to make sure things backwards compatible?

Reply:
Yes.

@nicholaspcr the protobuf binary encoding is breaking, an enum is a varint and it was a string. In JSON it's (probably) both a string, and there the values are only breaking indeed.

Reply:
Indeed maybe using constants on the backend and frontend to avoid breaking the API might be best.

We could do something like:
1. enforce that the string is within a list of acceptable values like we do [here](https://github.com/TheThingsNetwork/lorawan-stack/blob/d1f46cc41e6c75996bbddb61aeb68d8104cc7390/api/ttn/lorawan/v3/user.proto#L79-L91)
2. Keep the `enum NotificationType` but don't use it on the message definition.
3. Add TODO to replace the `string` type with the `NotificationType` on `v4`.
4. Enforce the usage of the `NotificationType.String()` internally on the back-end

We only have to enforce that they are either always on UPPERCASE or LOWERCASE, probably easier to add a temporary method on `ttnpb` to convert the defined `NotificationType` from upper case to lowercase.

So instead of doing 
```go
EmailTemplateName: ttnpb.NotificationType_USER_REQUESTED,
```

We do:
```go 
// Just converts to lowercase the string value of the enum type.
EmailTemplateName: ttnpb.GetNotificationType(ttnpb.NotificationType_USER_REQUESTED),
```

Reply:
Ok, we can use strings everywhere related to notifications, but still use enum for the user preferences, I think.
I think lowercase is better as well and we use a method to always convert